turbine/src/cluster_nodes.rs

```diff
@@ -451,15 +462,18 @@ fn get_seeded_rng(leader: &Pubkey, shred: &ShredId) -> ChaChaRng {
 // Each other node retransmits shreds to fanout many nodes in the next layer.
 // For example the node k in the 1st layer will retransmit to nodes:
 // fanout + k, 2*fanout + k, ..., fanout*fanout + k
+//
+// This function will only return Some(result) if `pred` returns true for at least
+// one of the items in `nodes`
 fn get_retransmit_peers<T>(
     fanout: usize,
     // Predicate fn which identifies this node in the shuffle.
     pred: impl Fn(T) -> bool,
     nodes: impl IntoIterator<Item = T>,
-) -> (/*this node's index:*/ usize, impl Iterator<Item = T>) {
+) -> Option<(/*this node's index:*/ usize, impl Iterator<Item = T>)> {
     let mut nodes = nodes.into_iter();
-    // This node's index within shuffled nodes.
-    let index = nodes.by_ref().position(pred).unwrap();
+    // This node's index should be somewhere within shuffled indices.
```

Comment:
I think we can keep the old comment

Reply:
addressed in [92202a0](https://github.com/anza-xyz/agave/pull/5380/commits/92202a07babf5e31fe1d1bc5cb585a539d770414)